torch/_decomp/decompositions.py

```diff
@@ -1218,3 +1218,60 @@ def stack(tensors: List[Tensor], dim: int = 0) -> Tensor:
         return out.view(result_sizes)
     else:
         return torch.cat(get_stack_inputs(tensors, wrapped_dim), dim)
+
+
+@register_decomposition(aten.frac)
+def frac(input: Tensor) -> Tensor:
+    return input - torch.trunc(input)
+
+
+@register_decomposition(aten.celu)
+def celu(input: Tensor, alpha: float = 1.0) -> Tensor:
+    inv_alpha = 1.0 / alpha
+    # TODO: Should I use F.elu here?
+    return aten.elu(input, alpha, 1.0, inv_alpha)
+
+
+@register_decomposition(aten.mish)
+@pw_cast_for_opmath
+def mish(x: Tensor) -> Tensor:
+    # TODO: copied from cuda impl, note this is different from doc's 'x * x.softplus().tanh()
+    return x * x.exp().log1p().tanh()
+
+
+@register_decomposition(aten.softplus)
+def softplus(a: Tensor, beta: float = 1.0, threshold: float = 20.0) -> Tensor:
+    return torch.where((a * beta) > threshold, a, (a * beta).exp().log1p() / beta)
+
+
+@register_decomposition(aten.softshrink)
+def softshrink(a: Tensor, lambd: float = 0.5) -> Tensor:
+    # TODO: can these two where be commbined?
+    return torch.where(a > lambd, a - lambd, torch.where(a < -lambd, a + lambd, 0))
+
+
+# TODO: why AOTAutograd tracing caught this? at::deg2rad has native impl
```

Comment:
@Chillee 

In UnaryOp.cpp, at::native::deg2rad() is implemented with at::mul_out(result, self, M_PI_180);
I am wondering why AOTAutograd tracer is not able further trace into its function body? 

Is it because AOTAutograd Tracer captures only at the aten op layer?

Reply:
It's because deg2rad is a compositeexplicitautograd op, and `__torch_dispatch__` traces above that level.

There *was* some discussion with @ezyang about whether we should make it possible to opt into CompositeExplicitAutograd decomps...

But in many cases, CompositeExplicitAutograd ops have things like out-variant ops, which make them somewhat unsuitable for decompositions.

Reply:
Got it. Thanks!